cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/MDSKey.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table;
+
+import co.cask.cdap.api.common.Bytes;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Metadata entry key
+ */
+public final class MDSKey {
+  private final byte[] key;
+
+  public MDSKey(byte[] key) {
```

Comment:
You should add Javadocs to all public methods. I am surprised that checkstyle is not failing - or is it? The Javadoc should state what exception is thrown in case the wrong type is requested, or if the end of the buffer is reached.